app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -2062,7 +2062,9 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
 
     @Override
     public void advance() {
-        next();
+        if (!questionsView.isQuestionList() && canNavigateForward()) {
```

Comment:
Oh also important to this- the condition you pointed out in `showNextView()` does not do what you thought it did (prevent navigating forward if it's on the last screen). I'm not sure why it doesn't, but when I remove the check for `canNavigateForward()`, the form gets submitted when an auto-advance question is on the last screen.